sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -1067,4 +1068,142 @@ public HttpPipeline getHttpPipeline() {
+        return new PagedFlux<>(
+            () -> queryFirstPage(query, clazz, context),
+            nextLink -> queryNextPage(query, nextLink, clazz, context));
+    }
+
+    Mono<PagedResponse<String>> queryFirstPage(String query, Context context) {
+        QuerySpecification querySpecification = new QuerySpecification();
+
+        querySpecification
+            .setQuery(query);
+
+        return protocolLayer
+            .getQueries()
+            .queryTwinsWithResponseAsync(querySpecification, context)
+            .map(objectPagedResponse -> new PagedResponseBase<>(
+                objectPagedResponse.getRequest(),
+                objectPagedResponse.getStatusCode(),
+                objectPagedResponse.getHeaders(),
+                objectPagedResponse.getValue().getItems().stream()
+                    .map(object -> {
+                        try {
+                            return mapper.writeValueAsString(object);
+                        } catch (JsonProcessingException e) {
+                            logger.error("JsonProcessingException occurred while retrieving query result items: ", e);
+                            throw new RuntimeException("JsonProcessingException occurred while retrieving query result items", e);
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()),
+                objectPagedResponse.getValue().getContinuationToken(),
+                objectPagedResponse.getDeserializedHeaders()));
+    }
+
+    <T> Mono<PagedResponse<T>> queryFirstPage(String query, Class<T> clazz, Context context) {
+        QuerySpecification querySpecification = new QuerySpecification();
+
+        querySpecification
+            .setQuery(query);
+
+        return protocolLayer
+            .getQueries()
+            .queryTwinsWithResponseAsync(querySpecification, context)
+            .map(objectPagedResponse -> new PagedResponseBase<>(
+                objectPagedResponse.getRequest(),
+                objectPagedResponse.getStatusCode(),
+                objectPagedResponse.getHeaders(),
+                objectPagedResponse.getValue().getItems().stream()
+                    .map(object -> mapper.convertValue(object, clazz))
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()),
+                objectPagedResponse.getValue().getContinuationToken(),
+                objectPagedResponse.getDeserializedHeaders()));
+    }
+
+    Mono<PagedResponse<String>> queryNextPage(String query, String nextLink, Context context) {
+        QuerySpecification querySpecification = new QuerySpecification();
+
+        querySpecification
+            .setQuery(query)
+            .setContinuationToken(nextLink);
+
+        return protocolLayer
+            .getQueries()
+            .queryTwinsWithResponseAsync(querySpecification, context)
```

Comment:
I see that the [query API](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/QueriesImpl.java#L72) doesn't return a pageable, is it the same for the .net client as well?

Reply:
yes it is the same for .net as well. We need to convert it to pageable.